libcaf_net/src/ip.cpp

```diff
@@ -0,0 +1,144 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2018 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "caf/config.hpp"
+#include "caf/detail/socket_sys_includes.hpp"
+#include "caf/error.hpp"
+#include "caf/ip_address.hpp"
+#include "caf/ipv4_address.hpp"
+#include "caf/logger.hpp"
+#include "caf/string_algorithms.hpp"
+#include "caf/string_view.hpp"
+
+// clang-format off
+#ifdef CAF_WINDOWS
+#  ifndef _WIN32_WINNT
+#    define _WIN32_WINNT 0x0600
+#  endif
+#  include <iostream>
+#  include <iphlpapi.h>
+#else
+#  include <net/if.h>
+#  include <netdb.h>
+#  include <ifaddrs.h>
+#  include <sys/ioctl.h>
+#  include <sys/types.h>
+#endif
+// clang-format on
+
+#ifndef HOST_NAME_MAX
+#  define HOST_NAME_MAX 255
+#endif
+
+using std::pair;
+using std::string;
+using std::vector;
+
+namespace caf {
+namespace net {
+namespace ip {
+
+namespace {
+
+template <class T>
+void* vptr(T* ptr) {
```

Comment:
This function is unnecessary. Casts from any pointer to `void*` are implicit in C++.